packages/account/imAccount/index.ts

```diff
@@ -0,0 +1,126 @@
+import type { BigNumberish, BytesLike, HexString } from "~typing"
+
+import { imAccountFactory } from "./factory"
+
+export class imAccount extends AccountSkeleton<imAccountFactory> {
+  /**
+   * Use when account is already deployed
+   */
+  public static async init(opts: {
+    address: string
+    nodeRpcUrl: string
+    validator: Validator
+    entryPointAddress: string
+  }) {
+    return new imAccount({
+      address: opts.address,
+      nodeRpcUrl: opts.nodeRpcUrl,
+      validator: opts.validator,
+      entryPointAddress: opts.entryPointAddress
+    })
+  }
+
+  /**
+   * Use when account is not yet deployed
+   */
+
+  public static async initWithFactory(opts: {
+    factoryAddress: string
+    implementationAddress: string
+    entryPointAddress: string
+    validator: Validator
+    fallbackHandlerAddress: string
+    salt: BigNumberish
+    nodeRpcUrl: string
+  }) {
+    const factory = new imAccountFactory({
+      factoryAddress: opts.factoryAddress,
+      implementationAddress: opts.implementationAddress,
+      entryPointAddress: opts.entryPointAddress,
+      validator: opts.validator,
+      fallbackHandlerAddress: opts.fallbackHandlerAddress,
+      salt: opts.salt,
+      nodeRpcUrl: opts.nodeRpcUrl
+    })
+    return new imAccount({
+      address: await factory.getAddress(),
+      factory,
+      nodeRpcUrl: opts.nodeRpcUrl,
+      validator: opts.validator,
+      entryPointAddress: opts.entryPointAddress
+    })
+  }
+
+  public validator: Validator
+  private account: ethers.Contract
+  private entryPointAddress: string
+
+  private constructor(opts: {
+    address: HexString
+    factory?: imAccountFactory
+    nodeRpcUrl: string
+    validator: Validator
+    entryPointAddress: string
+  }) {
+    super({
+      address: opts.address,
+      factory: opts.factory,
+      nodeRpcUrl: opts.nodeRpcUrl
+    })
+    this.account = new ethers.Contract(
+      opts.address,
+      imAccountMetadata.abi,
+      this.node
+    )
+    this.validator = opts.validator
+    this.entryPointAddress = opts.entryPointAddress
+  }
+
+  public changeValidator(newValidator: Validator) {
+    this.validator = newValidator
+  }
+
+  public async sign(message: BytesLike, metadata?: any) {
+    return await this.validator.sign(message, metadata)
+  }
+
+  protected async getCallData(call: Call): Promise<HexString> {
+    const calls: Call[] = [
+      {
+        to: call.to,
+        value: call.value ?? 0,
+        data: call.data ?? "0x"
+      }
+    ]
+    return this.account.interface.encodeFunctionData("execute", [calls])
+  }
+
+  protected async getDummySignature(): Promise<HexString> {
+    return ethers.AbiCoder.defaultAbiCoder().encode(
+      ["address", "bytes"],
+      [
+        await this.validator.getAddress(),
+        "0xfffffffffffffffffffffffffffffff0000000000000000000000000000000007aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa1c"
+      ]
+    )
```

Comment:
This should be put in validator, since only validator would know how to construct a valid signature for it.